gadopt/stokes_integrators.py

```diff
@@ -101,49 +102,41 @@ def __init__(self, z, T, approximation, bcs=None, mu=1,
                  quad_degree=6, cartesian=True, solver_parameters=None,
                  closed=True, rotational=False, J=None, constant_jacobian=False,
                  **kwargs):
+
         self.Z = z.function_space()
         self.mesh = self.Z.mesh()
         self.test = fd.TestFunctions(self.Z)
-        self.equations = StokesEquations(self.Z, self.Z, quad_degree=quad_degree,
-                                         compressible=approximation.compressible)
         self.solution = z
+
         self.approximation = approximation
+        self.bcs = bcs
         self.mu = ensure_constant(mu)
+        self.quad_degree = quad_degree
         self.solver_parameters = solver_parameters
         self.J = J
         self.constant_jacobian = constant_jacobian
-        self.linear = not depends_on(self.mu, self.solution)
 
+        self.linear = not depends_on(self.mu, self.solution)
         self.solver_kwargs = kwargs
-        u, p = fd.split(self.solution)
         self.k = upward_normal(self.Z.mesh(), cartesian)
+
+        # Add velocity, pressure and buoyancy term to the fields dictionary
+        self.stokes_vars = fd.split(self.solution)  # For default stokes this is a tuple of (velocity, pressure)
+
         self.fields = {
-            'velocity': u,
-            'pressure': p,
+            'velocity': self.stokes_vars[0],
+            'pressure': self.stokes_vars[1],
             'viscosity': self.mu,
             'interior_penalty': fd.Constant(2.0),  # allows for some wiggle room in imposition of weak BCs
                                                    # 6.25 matches C_ip=100. in "old" code for Q2Q1 in 2d.
-            'source': self.approximation.buoyancy(p, T) * self.k,
+            'source': self.approximation.buoyancy(self.stokes_vars[1], T) * self.k,
```

Comment:
Didn't we prefer `self.fields["pressure"]` to make it more explicit?

Reply:
Thanks for having a look over this thomas. When I actually ran this, the problem with this was that fields doesn't exist yet!

Reply:
Very true! Should be fine as is then :)